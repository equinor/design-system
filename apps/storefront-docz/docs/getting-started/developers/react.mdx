---
title: Developers
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
