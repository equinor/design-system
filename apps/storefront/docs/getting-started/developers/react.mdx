---
title: Developers
metaTitle: Developers react
tabs:
  - Introduction
  - CSS framework
  - React
  - Figma
mode: draft
route: /getting-started/developers/react/
---

:warning: Information coming soon
