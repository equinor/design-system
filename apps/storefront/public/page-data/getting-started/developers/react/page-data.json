{"componentChunkName":"component---docs-getting-started-developers-react-mdx","path":"/getting-started/developers/react/","result":{"pageContext":{"frontmatter":{"title":"Developers","metaTitle":"Developers react","tabs":["Introduction","CSS framework","React","Figma"],"mode":"draft","route":"/getting-started/developers/react/"},"entry":{"id":"6f9784da00517d57e4627269567073e6","filepath":"docs/getting-started/developers/react.mdx","fullpath":"/Users/vnys/Kode/storefront-replace/apps/storefront/docs/getting-started/developers/react.mdx","route":"/getting-started/developers/react/","slug":"docs-getting-started-developers-react","name":"React","menu":"","headings":[]}}}}