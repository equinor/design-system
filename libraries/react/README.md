@eds-ui/react
===============

1. `npm i -g pnpm`
1. `pnpm install`
1. `pnpm recursive install`

## Run storybook

`pnpm start`

## Build modules

`pnpm recursive run build`
