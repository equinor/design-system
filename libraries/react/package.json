{
  "name": "@eds-ui/react",
  "private": true,
  "version": "0.0.1",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "jest",
    "storybook": "start-storybook"
  },
  "keywords": [],
  "author": "",
  "license": "AGPL",
  "dependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6"
  },
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.4",
    "@babel/preset-env": "^7.4.4",
    "@babel/preset-react": "^7.0.0",
    "@storybook/react": "^5.0.11",
    "babel-plugin-styled-components": "^1.10.0",
    "enzyme": "^3.9.0",
    "enzyme-adapter-react-16": "^1.12.1",
    "jest": "^24.8.0",
    "jest-styled-components": "^6.3.1",
    "rollup": "^1.11.3",
    "rollup-plugin-babel": "^4.3.2",
    "rollup-plugin-commonjs": "^9.3.4",
    "rollup-plugin-node-resolve": "^4.2.3",
    "styled-components": "^4.2.0"
  },
  "peerDependencies": {
    "react": "^16.8.6",
    "react-dom": "^16.8.6",
    "styled-components": "^4.2.0",
    "prop-types": "^15.7.2"
  },
  "jest": {
    "verbose": true,
    "setupFilesAfterEnv": [
      "<rootDir>/testSetup.js"
    ],
    "testPathIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "transformIgnorePatterns": [
      "<rootDir>/(node_modules)"
    ],
    "haste": {
      "providesModuleNodeModules": [
        ".*"
      ]
    },
    "roots": [
      "packages/node_modules"
    ]
  }
}
