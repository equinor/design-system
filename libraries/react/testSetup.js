import React from 'react'
import Enzyme from 'enzyme'
import 'jest-styled-components'
import styled from 'styled-components'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })
