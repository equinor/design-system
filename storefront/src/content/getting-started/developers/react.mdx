---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
mode: draft
---


:warning: Information coming soon
