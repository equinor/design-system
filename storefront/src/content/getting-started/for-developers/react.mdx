---
title: React
tabs:
    - Introduction
    - CSS framework
    - React
    - Figma
    - Contributing
---

Developers React
