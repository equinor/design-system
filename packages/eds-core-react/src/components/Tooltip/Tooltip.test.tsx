/* eslint-disable no-undef */
import { fireEvent, render, screen, act } from '@testing-library/react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import { axe } from 'jest-axe'
import styled from 'styled-components'
import { Tooltip, Button } from '../../'

const StyledTooltip = styled(Tooltip)`
  background: red;
`
const openDelay = 100

const mockResizeObserver = jest.fn(() => ({
  observe: jest.fn(),
  disconnect: jest.fn(),
  unobserve: jest.fn(),
}))

beforeAll(() => {
  window.ResizeObserver = mockResizeObserver
})

describe('Tooltip', () => {
  it('Matches snapshot', async () => {
    render(
      <Tooltip title="Tooltip" enterDelay={0}>
        <span>Test</span>
      </Tooltip>,
    )

    const content = screen.getByText('Test')
    fireEvent.mouseEnter(content)
    const tooltip = await screen.findByRole('tooltip')
    expect(tooltip).toMatchSnapshot()
  })
  it('Should pass a11y test', async () => {
    const { container } = render(
      <Tooltip title="Tooltip">
        <span>Test</span>
      </Tooltip>,
    )

    const content = screen.getByText('Test')

    fireEvent.mouseEnter(content)
    await act(() => new Promise((r) => setTimeout(r, openDelay)))

    const results = await axe(container)

    expect(results).toHaveNoViolations()
  })
  it('Should pass a11y test with id & placement', async () => {
    const { container } = render(
      <Tooltip title="Tooltip" id="a11y-tooltip" placement="top" enterDelay={0}>
        <span>Test</span>
      </Tooltip>,
    )

    const content = screen.getByText('Test')

    fireEvent.mouseEnter(content)

    await act(async () => {
      expect(await axe(container)).toHaveNoViolations()
    })
  })
  it('can extend the css for the component', async () => {
    render(
      <StyledTooltip title="Tooltip">
        <span>Test</span>
      </StyledTooltip>,
    )

    const content = screen.getByText('Test')

    fireEvent.mouseEnter(content)
    await act(() => new Promise((r) => setTimeout(r, openDelay)))

    const tooltip = await screen.findByRole('tooltip')
    expect(tooltip).toHaveStyleRule('background', 'red')
  })
  it('is visible when content is being hovered', async () => {
    render(
      <Tooltip title="Tooltip" placement="right-start">
        <div>Test</div>
      </Tooltip>,
    )

    const content = screen.getByText('Test')

    fireEvent.mouseEnter(content)
    await act(() => new Promise((r) => setTimeout(r, openDelay)))

    const tooltip = await screen.findByRole('tooltip')
    expect(tooltip).toBeInTheDocument()
  })
  it('renders with a correct title', async () => {
    render(
      <Tooltip title="Tooltip" placement="right-start">
        <span>Test</span>
      </Tooltip>,
    )

    const content = screen.getByText('Test')

    fireEvent.mouseEnter(content)
    await act(() => new Promise((r) => setTimeout(r, openDelay)))

    const tooltip = await screen.findByRole('tooltip')

    expect(content).toBeDefined()
    expect(tooltip).toBeDefined()
  })
  it('renders on focus', async () => {
    render(
      <Tooltip title="Tooltip" placement="right-start">
        <span>Test</span>
      </Tooltip>,
    )

    const content = screen.getByText('Test')

    fireEvent.focusIn(content)
    await act(() => new Promise((r) => setTimeout(r, openDelay)))

    expect(await screen.findByRole('tooltip')).toBeDefined()
  })
  it('shows after correct delay', async () => {
    render(
      <Tooltip title="Tooltip" placement="right-start" enterDelay={500}>
        <span>Test</span>
      </Tooltip>,
    )

    const content = screen.getByText('Test')

    fireEvent.mouseEnter(content)
    await act(() => new Promise((r) => setTimeout(r, 200)))
    expect(screen.queryByText('Tooltip')).not.toBeInTheDocument()

    await act(() => new Promise((r) => setTimeout(r, 300)))

    const tooltip = await screen.findByRole('tooltip')

    expect(content).toBeDefined()
    expect(tooltip).toBeDefined()
  })
  it('child onFocus is called when focused', async () => {
    const handler = jest.fn()
    render(
      <Tooltip title="Tooltip">
        <Button onFocus={handler}>Test</Button>
      </Tooltip>,
    )

    const button = screen.getByText('Test')

    fireEvent.focus(button)
    await act(() => new Promise((r) => setTimeout(r, openDelay)))

    expect(handler).toBeCalled()
  })
})
