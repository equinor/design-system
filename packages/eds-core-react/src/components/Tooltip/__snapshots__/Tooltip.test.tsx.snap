// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip Matches snapshot 1`] = `
.c0 {
  margin: 0;
  color: var(--eds_text__static_icons__primary_white,rgba(255,255,255,1));
  font-family: Equinor;
  font-size: 0.750rem;
  font-weight: 500;
  line-height: 1.333em;
  text-align: left;
  padding-left: 8px;
  padding-top: 8px;
  padding-right: 8px;
  padding-bottom: 8px;
  border-radius: 4px;
  background: var(--eds_ui_background__overlay,rgba(0,0,0,0.8));
  z-index: 1500;
  white-space: nowrap;
}

.c1 {
  position: absolute;
  width: 6px;
  height: 8px;
  z-index: -1;
}

.c2 {
  width: 6px;
  height: 8px;
  position: absolute;
  fill: var(--eds_ui_background__overlay,rgba(0,0,0,0.8));
}

<div
  class="c0"
  id="floating-ui-0"
  role="tooltip"
  style="position: absolute; top: 14px; left: 8px;"
  tabindex="-1"
>
  Tooltip
  <div
    class="c1"
    style="left: 0px; top: -6px; transform: rotate(90deg);"
  >
    <svg
      class="c2 arrowSvg"
    >
      <path
        d="M0.504838 4.86885C-0.168399 4.48524 -0.168399 3.51476 0.504838 3.13115L6 8.59227e-08L6 8L0.504838 4.86885Z"
      />
    </svg>
  </div>
</div>
`;
